contracts/fixed_price_oracle.vy

```diff
@@ -0,0 +1,100 @@
+struct PriceInfo:
+    price: uint256
+    lastUpdate: uint256
+
+contract ERC20():
+    def balanceOf(_owner: address) -> uint256: constant
+    def decimals() -> uint256: constant
+
+contract DaiPriceOracle:
+    def g_priceInfo() -> PriceInfo: constant
+
+PRICE_MULTIPLIER: constant(uint256) = 100000000 # 10**8
+
+# DAI oracle returns DAI/USD price multiplied by 10**18
+MIN_DAI_PRICE: constant(uint256) = 950000000000000000 # 0.95*10**18
+MAX_DAI_PRICE: constant(uint256) = 1050000000000000000 # 1.05*10**18
+# DAI_ORACLE_DIVIDER = 10**18 / PRICE_MULTIPLIER
+DAI_ORACLE_DIVIDER: constant(uint256) = 10000000000 # 10**10
+
+TokenAddressUpdated: event({token_address: indexed(address), token_index: indexed(int128)})
+
+name: public(string[16])
+owner: public(address)
+supportedTokens: public(address[5])
+allowedTokens: public(map(address, bool))
+daiAddress: public(address)
+daiOracleAddress: public(address)
+
+@public
+def __init__(dai_address: address, dai_oracle_address: address):
+    assert dai_address != ZERO_ADDRESS
+    assert dai_oracle_address != ZERO_ADDRESS
+
+    self.daiAddress = dai_address
+    self.daiOracleAddress = dai_oracle_address
+    self.allowedTokens[dai_address] = True
+    self.supportedTokens[0] = dai_address
+    self.owner = msg.sender
+    self.name = 'PriceOracle'
+    
+@private
+@constant
+def _normalized_token_price(token_address: address) -> uint256:
```

Comment:
What's the motivation for moving this into a private function and not having this code only in the public function without the _ prefix?

Reply:
The motivation is to make it compatible with vyper 0.1.0b12 and future versions. b12 doesn't allow local calls of public functions, but we call this function from ```poolSize```.

Reply:
I see. For now, we don't need to support b12 and probably won't need to for a long time until we upgrade the contract. Let's remove the prefix.

Reply:
Done